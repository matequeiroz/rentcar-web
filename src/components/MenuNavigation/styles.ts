import styled from "@emotion/styled";

export const Container = styled.div`
    background-color: var(--gray-200);
    position: fixed;
    bottom: 0;
    width: 100%;
    height: 7rem;

    nav {
        margin: 0 4.9rem;
        height: inherit;
        
        ul {
            display: flex;
            justify-content: space-between;
            align-items: center;
            height: inherit;
            list-style: none;

            svg {
                color: var(--gray-300);
            }
        }
    }
`