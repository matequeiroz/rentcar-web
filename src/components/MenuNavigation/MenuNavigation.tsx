import { BiHomeAlt2, BiCar, BiUser } from "react-icons/bi";

import * as S from './styles'

export default function MenuNavigation() {
  return (
    <S.Container>
      <nav data-testid="MenuNavigation-component">
        <ul>
          <li data-testid="MenuNavigation-item__component">
            <BiHomeAlt2 size={22} />
          </li>
          <li data-testid="MenuNavigation-item__component">
            <BiCar size={22} />
          </li>
          <li data-testid="MenuNavigation-item__component"> 
            <BiUser size={22} />
          </li>
        </ul>
      </nav>
    </S.Container>
  );
}
