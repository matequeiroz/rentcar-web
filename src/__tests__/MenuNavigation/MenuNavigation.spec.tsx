import { render, screen } from "@testing-library/react";

import MenuNavigation  from "@/components/MenuNavigation/MenuNavigation";

describe("MenuNavigation.tsx", () => {

  it("should render the component", () => {
    const { container } = render(<MenuNavigation />);
    expect(screen.getAllByTestId('MenuNavigation-component').length).toBe(1);
    expect(screen.getAllByTestId("MenuNavigation-item__component").length).toBe(3)
    expect(container).toMatchSnapshot()
  });
});
