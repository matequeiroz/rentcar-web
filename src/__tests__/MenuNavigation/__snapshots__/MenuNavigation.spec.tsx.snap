// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuNavigation.tsx should render the component 1`] = `
<div>
  <div
    class="css-12hxdqe"
  >
    <nav
      data-testid="MenuNavigation-component"
    >
      <ul>
        <li
          data-testid="MenuNavigation-item__component"
        >
          <svg
            fill="currentColor"
            height="22"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.71 2.29a1 1 0 0 0-1.42 0l-9 9a1 1 0 0 0 0 1.42A1 1 0 0 0 3 13h1v7a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-7h1a1 1 0 0 0 1-1 1 1 0 0 0-.29-.71zM6 20v-9.59l6-6 6 6V20z"
            />
          </svg>
        </li>
        <li
          data-testid="MenuNavigation-item__component"
        >
          <svg
            fill="currentColor"
            height="22"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m20.772 10.156-1.368-4.105A2.995 2.995 0 0 0 16.559 4H7.441a2.995 2.995 0 0 0-2.845 2.051l-1.368 4.105A2.003 2.003 0 0 0 2 12v5c0 .753.423 1.402 1.039 1.743-.013.066-.039.126-.039.195V21a1 1 0 0 0 1 1h1a1 1 0 0 0 1-1v-2h12v2a1 1 0 0 0 1 1h1a1 1 0 0 0 1-1v-2.062c0-.069-.026-.13-.039-.195A1.993 1.993 0 0 0 22 17v-5c0-.829-.508-1.541-1.228-1.844zM4 17v-5h16l.002 5H4zM7.441 6h9.117c.431 0 .813.274.949.684L18.613 10H5.387l1.105-3.316A1 1 0 0 1 7.441 6z"
            />
            <circle
              cx="6.5"
              cy="14.5"
              r="1.5"
            />
            <circle
              cx="17.5"
              cy="14.5"
              r="1.5"
            />
          </svg>
        </li>
        <li
          data-testid="MenuNavigation-item__component"
        >
          <svg
            fill="currentColor"
            height="22"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
            />
          </svg>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
